tahrir/views/report.py

```diff
@@ -135,61 +135,63 @@ def report_this_month():
-                raise ValueError()
-        elif month is not None:
-            # Monthly report
-            start = date(year, month, 1)
-            stop = start + timedelta(days=32)
-            stop = stop.replace(day=1)
-            stop = stop - timedelta(days=1)
-        elif year is not None:
-            # Yearly report
-            start = date(year, 1, 1)
-            stop = date(year, 12, 31)
+    begin = int(request.args.get('begin', 0))
+    limit = int(request.args.get('limit', 200))
+    if request.path.endswith('/week'):
+        # Weekly report
+        start = get_start_week(year, month, day)
+        stop = start + timedelta(days=6)
+        print("Start: ", start, " End: ", stop)
+    elif day is not None:
+        # Daily report
+        start = date(year, month, day)
+        stop = start + timedelta(days=1)
+    elif week is not None:
+        # Weekly report
+        max_week = date(year, 12, 31).isocalendar()[1]
+        if 0 < week < max_week:
+            start = date(year, 1, 1) + timedelta(weeks=week - 1)
+            start = get_start_week(start.year, start.month, start.day)
+            stop = start + timedelta(days=6)
         else:
-            # Default: Complete list
-            start = None
-            stop = None
-
-        # TODO: Modify make_leaderboard to add filter
-        user_to_rank = g.tahrirdb.make_leaderboard(
-            start=start,
-            stop=stop,
-        )
-
-        limited_users = list(user_to_rank)[begin:begin+limit]
-
-        data = [
-            {
-                "email": user.email,
-                "id": user.id,
-                "nickname": user.nickname,
-                "avatar": get_avatar(user.avatar, int(request.args.get("size", 100))),
-                "rank": user.rank,
-                "period_badges": user_to_rank[user]["badges"]
-            } for user in limited_users
-        ]
-        return jsonify(data)
-    except ValueError:
-        abort(422,"Incorrect system input!")
+            raise ValueError()
+    elif month is not None:
+        # Monthly report
+        start = date(year, month, 1)
+        stop = start + timedelta(days=32)
+        stop = stop.replace(day=1)
+        stop = stop - timedelta(days=1)
+    elif year is not None:
+        # Yearly report
+        start = date(year, 1, 1)
+        stop = date(year, 12, 31)
+    else:
+        # Default: Complete list
+        start = None
+        stop = None
+
+    # TODO: Modify make_leaderboard to add filter
+    user_to_rank = g.tahrirdb.make_leaderboard(
+        start=start,
+        stop=stop,
+    )
+
+    limited_users = list(user_to_rank)[begin:begin+limit]
+
+    data = [
+        {
+            "mail": user.avatar,
+            "nickname": user.nickname,
+            "badges": user_to_rank[user]["badges"],
+            "rank": {
+                "global": user.rank,
+                "period": user_to_rank[user]["rank"],
+            },
+        } for user in limited_users
```

Comment:
This is an API change, but I don't think many people are using the badges API (un)fortunately.
It would be useful to also add the `user.id` to have a unique reference for the user. In React, it can be used for the `key` attribute [as recommended by the docs](https://react.dev/learn/rendering-lists#where-to-get-your-key).